.changeset/forty-trains-notice.md

```diff
@@ -0,0 +1,7 @@
+---
+'astro': minor
+---
+
+Makes `defineConfig` generic
```

Comment:
I'm not sure we should use this phrase. There isn't any generic that **the user should provide**. 

I would rephrase the changeset in a way that highlights the auto completion and the type safety.